Sources/Brave/Frontend/Sync/SyncSettingsTableViewController.swift

```diff
@@ -82,112 +140,101 @@ class SyncSettingsTableViewController: UIViewController, UITableViewDelegate, UI
 
     switch type {
     case .lastDeviceLeft:
       title = String(format: Strings.syncRemoveLastDeviceTitle, deviceName)
       message = Strings.syncRemoveLastDeviceMessage
       removeButtonName = Strings.syncRemoveLastDeviceRemoveButtonName
-    case .currentDevice:
+    case .currentDeviceLeft:
       title = String(format: Strings.syncRemoveCurrentDeviceTitle, "\(deviceName) (\(Strings.syncThisDevice))")
       message = Strings.syncRemoveCurrentDeviceMessage
       removeButtonName = Strings.removeDevice
-    case .otherDevice:
+    case .otherDeviceLeft:
       title = String(format: Strings.syncRemoveOtherDeviceTitle, deviceName)
       message = Strings.syncRemoveOtherDeviceMessage
       removeButtonName = Strings.removeDevice
+    case .syncChainDeleteConfirmation:
+      title = Strings.Sync.syncDeleteAccountAlertTitle
+      message = "\(Strings.Sync.syncDeleteAccountAlertDescriptionPart1).\n\n\(Strings.Sync.syncDeleteAccountAlertDescriptionPart2)"
+      removeButtonName = Strings.delete
+    case .syncChainDeleteError:
+      title = Strings.Sync.syncChainAccountDeletionErrorTitle
+      message = Strings.Sync.syncChainAccountDeletionErrorDescription
+      removeButtonName = Strings.OKString
     }
 
     guard let popupTitle = title, let popupMessage = message, let popupButtonName = removeButtonName else { fatalError() }
 
-    let popup = AlertPopupView(imageView: nil, title: popupTitle, message: popupMessage)
-    let fontSize: CGFloat = 15
+    let popup = AlertPopupView(title: popupTitle, message: popupMessage)
 
-    popup.addButton(title: Strings.cancelButtonTitle, fontSize: fontSize) { return .flyDown }
-    popup.addButton(title: popupButtonName, type: .destructive, fontSize: fontSize) {
+    popup.addButton(title: Strings.cancelButtonTitle) { return .flyDown }
+    popup.addButton(title: popupButtonName, type: .destructive) {
       switch type {
-      case .lastDeviceLeft, .currentDevice:
+      case .lastDeviceLeft, .currentDeviceLeft:
         self.syncAPI.leaveSyncGroup()
         self.navigationController?.popToRootViewController(animated: true)
-      case .otherDevice:
-        self.syncAPI.removeDeviceFromSyncGroup(deviceGuid: device.guid)
+      case .otherDeviceLeft:
+        if let guid = device?.guid {
+          self.syncAPI.removeDeviceFromSyncGroup(deviceGuid: guid)
+        }
+      case .syncChainDeleteConfirmation:
+        self.doIfConnected {
+          // Observers have to be removed before permanentlyDeleteAccount is called
+          // to prevent glitch in settings screen
+          self.syncAPI.removeAllObservers()
+          // Start  loading and diable navigation while delete action is happening
+          self.enableNavigationPrevention()
+          
+          // Permanently Delete action is called on brave-core side
+          self.syncAPI.permanentlyDeleteAccount { [weak self] status in
+            guard let self else { return }
```

Comment:
You may want to keep `self` alive or ensure that 
`syncAPI.leaveSyncGroup` is still called if `self` dies. Given that you are padding this with `enableNavigationPrevention`/`disableNavigationPrevention` it is safe to assume you can just avoid weak self here right?

You could also capture `syncAPI` explicitly in the capture list so that you can still execute the `default` switch case

Reply:
Fixed at https://github.com/brave/brave-ios/pull/7031/commits/3d172a29f0a8d62dacb9675594e1f0f30ae29ee7